devsetup/Makefile

```diff
@@ -83,6 +84,41 @@ crc_cleanup: ## Destroys the CRC env, but does NOT clear ( --clear-cache ) the c
 	sudo ${CLEANUP_DIR_CMD} /etc/pki/ca-trust/source/anchors/crc-router-ca.pem
 	sudo update-ca-trust
 
+.PHONY: add_nics_crc
+add_nics_crc: remove_nics_crc
+	# Add data interface
+	MAC_ADDRESS=$(shell echo -n 52:54:00; dd bs=1 count=3 if=/dev/random 2>/dev/null | hexdump -v -e '/1 "-%02X"' | tr '-' ':'); \
+	virsh --connect=qemu:///system attach-interface --domain crc --type network --source data --model virtio --mac $$MAC_ADDRESS --config;
+	sleep 1
+	# Add management interface
+	MAC_ADDRESS=$(shell echo -n 52:54:00; dd bs=1 count=3 if=/dev/random 2>/dev/null | hexdump -v -e '/1 "-%02X"' | tr '-' ':'); \
+	virsh --connect=qemu:///system attach-interface --domain crc --type network --source management --model virtio --mac $$MAC_ADDRESS --config;
+	sleep 1
+	# Add external interface
+	MAC_ADDRESS=$(shell echo -n 52:54:00; dd bs=1 count=3 if=/dev/random 2>/dev/null | hexdump -v -e '/1 "-%02X"' | tr '-' ':'); \
+	virsh --connect=qemu:///system attach-interface --domain crc --type network --source external --model virtio --mac $$MAC_ADDRESS --config;
+	sleep 1
+	virsh --connect=qemu:///system destroy crc
+	timeout --foreground ${TIMEOUT_VALUE} bash -c 'until $$(virsh --connect=qemu:///system list --all | grep crc | grep -q "shut off"); do sleep 1; done'
+	virsh --connect=qemu:///system start crc
+	timeout --foreground ${TIMEOUT_VALUE} bash -c 'until $$(virsh --connect=qemu:///system list --all | grep crc | grep -q "running"); do sleep 1; done'
+
+.PHONY: remove_nics_crc
+remove_nics_crc:
+	-MAC_ADDRESS=$(shell virsh --connect=qemu:///system dumpxml crc | grep "source network='data'" -B1 | grep mac | cut -d"'" -f2); \
```

Comment:
this should probably be a loop?

Reply:
I find it harder than expected to create a loop on the Makefile itself where every iteration you need to assign a new variable.
For example, given the same code as add_crc_nics but using loop:
```
.PHONY: test_add_nics_crc
test_add_nics_crc: test_rem_nics_crc
        for network in data management external; do \
                MAC_ADDRESS=$(shell echo -n 52:54:00; dd bs=1 count=3 if=/dev/random 2>/dev/null | hexdump -v -e '/1 "-%02X"' | tr '-' ':'); \
                echo "virsh --connect=qemu:///system attach-interface --domain crc --type network --source $$network --model virtio --mac $$MAC_ADDRESS --config;"; \
                sleep 1; \
        done
```
When executed this will happen:
```

for network in data management external; do \
        MAC_ADDRESS=52:54:00:13:72:0E; \
        echo "virsh --connect=qemu:///system attach-interface --domain crc --type network --source $network --model virtio --mac $MAC_ADDRESS --config;"; \                                                                                  
        sleep 1; \
done
virsh --connect=qemu:///system attach-interface --domain crc --type network --source data --model virtio --mac 52:54:00:13:72:0E --config;                                                                                                   
virsh --connect=qemu:///system attach-interface --domain crc --type network --source management --model virtio --mac 52:54:00:13:72:0E --config;                                                                                             
virsh --connect=qemu:///system attach-interface --domain crc --type network --source external --model virtio --mac 52:54:00:13:72:0E --config;                                                                                               
```
I agree that both targets (add/remove nics) could be handled using a loop, but I guess the best way should be running an script

Reply:
A script would be nice I think